contracts/modules/TransferManager.sol

```diff
@@ -0,0 +1,532 @@
+        public
+        onlyWhenUnlocked(_wallet)
+    {
+        bytes32 id = keccak256(abi.encodePacked(ActionType.Transfer, _token, _to, _amount, _data, _block));
+        uint executeAfter = configs[address(_wallet)].pendingActions[id];
+        require(executeAfter > 0, "TT: unknown pending transfer");
+        uint executeBefore = executeAfter.add(securityWindow);
+        require(executeAfter <= now && now <= executeBefore, "TT: transfer outside of the execution window");
+        delete configs[address(_wallet)].pendingActions[id];
+        doTransfer(_wallet, _token, _to, _amount, _data);
+        emit PendingTransferExecuted(address(_wallet), id);
+    }
+
+    function cancelPendingTransfer(
+        BaseWallet _wallet,
+        bytes32 _id
+    )
+        public
+        onlyWalletOwner(_wallet)
+        onlyWhenUnlocked(_wallet)
+    {
+        require(configs[address(_wallet)].pendingActions[_id] > 0, "TT: unknown pending action");
+        delete configs[address(_wallet)].pendingActions[_id];
+        emit PendingTransferCanceled(address(_wallet), _id);
+    }
+
+    /**
+     * @dev Lets the owner of a wallet change its global limit.
+     * The limit is expressed in ETH. Changes to the limit take 24 hours.
+     * @param _wallet The target wallet.
+     * @param _newLimit The new limit.
+     */
+    function changeLimit(BaseWallet _wallet, uint256 _newLimit) public onlyWalletOwner(_wallet) onlyWhenUnlocked(_wallet) {
+        changeLimit(_wallet, _newLimit, securityPeriod);
+    }
+
+    /**
+     * @dev Convenience method to disable the limit
+     * The limit is disabled by setting it to an arbitrary large value.
+     * @param _wallet The target wallet.
+     */
+    function disableLimit(BaseWallet _wallet) external onlyWalletOwner(_wallet) onlyWhenUnlocked(_wallet) {
+        changeLimit(_wallet, LIMIT_DISABLED, securityPeriod);
+    }
+
+    /**
+    * @dev Checks if an address is whitelisted for a wallet.
+    * @param _wallet The target wallet.
+    * @param _target The address.
+    * @return true if the address is whitelisted.
+    */
+    function isWhitelisted(BaseWallet _wallet, address _target) public view returns (bool _isWhitelisted) {
+        uint whitelistAfter = transferStorage.getWhitelist(_wallet, _target);
+        // solium-disable-next-line security/no-block-members
+        return whitelistAfter > 0 && whitelistAfter < now;
+    }
+
+    /**
+    * @dev Gets the info of a pending transfer for a wallet.
+    * @param _wallet The target wallet.
+    * @param _id The pending transfer ID.
+    * @return the epoch time at which the pending transfer can be executed.
+    */
+    function getPendingTransfer(BaseWallet _wallet, bytes32 _id) external view returns (uint64 _executeAfter) {
+        _executeAfter = uint64(configs[address(_wallet)].pendingActions[_id]);
+    }
+
+    /**
+    * @dev Implementation of EIP 1271.
+    * Should return whether the signature provided is valid for the provided data.
```

Comment:
I would say "should return a magic value if the provided signature is valid or revert otherwise".